pkg/storage/s3/s3.go

```diff
@@ -46,11 +50,23 @@ func NewDriver(c *imageregistryv1.ImageRegistryConfigStorageS3, listers *regopcl
 // getS3Service returns a client that allows us to interact
 // with the aws S3 service
 func (d *driver) getS3Service() (*s3.S3, error) {
+	ctx := context.TODO()
+
 	if s3Service != nil {
 		return s3Service, nil
 	}
 
-	cfg, err := clusterconfig.GetAWSConfig(d.Listers)
+	config, err := regopclient.GetConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := rest.RESTClientFor(config)
```

Comment:
There were a number of places that initialized fresh configs and clients from scratch.  It seems like we should open a single REST client early on (in `cmd/cluster-image-registry-operator`?) and then pass that around for these other functions to use.  That would also make it easy to plug in dummy clients for unit tests, etc.  But I wanted some feedback on that before pushing it though.  For now, I'm just inlining the various client initialization steps where we need them.